README.md

```diff
@@ -23,25 +23,135 @@ import {
 } from 'react-native';
 import InfiniteScrollView from 'react-native-infinite-scroll-view';
 
+class ExampleComponent extends React.Component {
+  _loadMoreContentAsync = async () => {
+    // Fetch more data here.
+    // After fetching data, you should update your ListView data source
+    // manually.
+    // This function does not have a return value.
+  }
 
-// Inside of a component's render() method:
-render() {
-  return (
-    <ListView
-      renderScrollComponent={props => <InfiniteScrollView {...props} />}
-      dataSource={...}
-      renderRow={...}
-      canLoadMore={this.state.canLoadMoreContent}
-      onLoadMoreAsync={this._loadMoreContentAsync()}
-    />
-  );
+  render() {
+    return (
+      <ListView
+        renderScrollComponent={props => <InfiniteScrollView {...props} />}
+        dataSource={...}
+        renderRow={...}
+        canLoadMore={this.state.canLoadMoreContent}
+        onLoadMoreAsync={this._loadMoreContentAsync}
+      />
+    );
+  }
 }
 ```
 
+A more complete example that uses a `ListView.DataSource`, [react-redux](https://github.com/reactjs/react-redux), and supports pagination would look something like this:
+
+```js
+import React from 'react';
+import {
+  ListView,
+  RefreshControl,
+} from 'react-native';
+import InfiniteScrollView from 'react-native-infinite-scroll-view';
+import { connect } from 'react-redux';
+
+class ExampleComponent extends React.Component {
+  static propTypes = {
+    // Assume data shape looks like:
+    // {items: ["item1", "item2"], nextUrl: null, isFetching: false}
+    listData: PropTypes.object.isRequired,
+
+    // dispatch is automatically provided by react-redux, and is used to
+    // interact with the store.
+    dispatch: PropTypes.object.isRequired,
```

Comment:
This should be PropTypes.func.isRequired, I don't think there is a more precise prop type though. Longer term btw the vision is to remove propTypes and use Flow types instead, potentially with a Babel plugin to check Flow types at runtime too.